### DISCUSSION TITLE
Question about coroutine, possible cause of context leak.

### DISCUSSION THREAD

**Discussion by deflinhec** [yuin/gopher-lua#437] [Q&A] [answered]
I'm aware of the LState NewThread function propagate a new context which never be cancel. 

https://github.com/yuin/gopher-lua/blob/d212719c8304b1dfbe24bbc86780668ac612fc94/state.go#L1563

https://github.com/yuin/gopher-lua/blob/d212719c8304b1dfbe24bbc86780668ac612fc94/coroutinelib.go#L21

I'm current working on a service which use the latest version of gopher-lua, according to the graph, which generate from /pprof/heap, It seems to be a context leak with the coroutine module?

![profile004-markout](https://github.com/yuin/gopher-lua/assets/1531222/fd578116-b74b-46de-a2a0-e72a78028879)

12 hours Memory usage deployed on Kubernetes, service get restarts after a peroid of time.

<img width="1080" alt="Screenshot 2023-05-18 at 13 26 32" src="https://github.com/yuin/gopher-lua/assets/1531222/8cd0ee2c-1995-4c34-a8fc-16571e76c9e4">

**Comment by deflinhec** [comment, marked as answer]
PR https://github.com/yuin/gopher-lua/pull/438, open for discussions.

**Reply by yuin** [reply]
Thanks, I've merged your PR.

**Comment by deflinhec** [comment]
By the way, I'm using [underscore.lua](https://github.com/mirven/underscore.lua).
